### DISCUSSION TITLE
select2_from_ajax / relationship - attribute made up of first_name, last_name

### DISCUSSION THREAD

**Discussion by sammyblackbaron** [Laravel-Backpack/community-forum#472] [Q&A (Help)] [answered]
tldr - how can I use select2_from_ajax / relationship / fetch with a model with multiple fields (e.g. first_name, last_name) that make up the name?

Hi,

I have a Contact model with first_name & last_name fields.  In another CrudController, I would like to use either a select2_from_ajax field type or a relationship field type (I've tried both).

I can get it to show the results in the select2 ajax search:

<img width="973" alt="Screenshot 2023-04-13 at 10 18 21" src="https://user-images.githubusercontent.com/3033242/231714620-809487ae-1469-419c-9979-5b551a9c6cef.png">

And it populates the field when I click on the option to select:

<img width="973" alt="Screenshot 2023-04-13 at 10 18 35" src="https://user-images.githubusercontent.com/3033242/231714690-8dbae30d-7fd2-4f81-9019-50bec612db21.png">

It even saves it to the database correctly.  However when I go back into the record, it doesn't show anything in the field - just the "clear" option:

<img width="973" alt="Screenshot 2023-04-13 at 10 18 47" src="https://user-images.githubusercontent.com/3033242/231714790-c9b241a0-b4ce-4582-96ff-d08629d2b2fa.png">

This is my code:

```php
$this->crud->addField([
    'type'          => "relationship",
    'name'          => 'contact',
    'ajax'          => true,
    'label'         => 'Contact',
    'attribute'     => 'name',
    'data_source' => backpack_url('user/fetch-contacts')
]);

protected function fetchContacts()
{
    return $this->fetch([
        'model' => \App\Models\Contact::class,
        'query' => function ($model) {
            return $model->with('organisation')
                ->selectRaw("contacts.id, CONCAT(organisations.name, ' - ', COALESCE(contacts.title, ''), ' ', COALESCE(contacts.first_name, ''), ' ', COALESCE(contacts.last_name, '')) as name, organisations.name as org_name")
                ->leftJoin('organisations', 'organisations.id', '=', 'contacts.organisation_id')
                ->orderByRaw('organisations.name ASC, contacts.last_name ASC');
        },
        'searchable_attributes' => ['first_name', 'last_name'],
        'distinct' => true,
    ]);
}
```

Can anyone tell me what I'm doing wrong?  Many thanks!

**Comment by o15a3d4l11s2** [comment, marked as answer]
**Update**

I remember a similar case a year ago. If I remember correctly the issue is that the keys are not taken in consideration for the filtering. As a quick test you can do the following:
```php
protected function fetchContacts()
{
    if (request()->get('keys')) {
        return $this->fetch([
            'model' => \App\Models\Contact::class,
            'searchable_attributes' => ['first_name', 'last_name'],
        ]);
    } else {
        // your current logic
        return $this->fetch([
            'model' => \App\Models\Contact::class,
            'query' => function ($model) {
                return $model->with('organisation')
                    ->selectRaw("contacts.id, CONCAT(organisations.name, ' - ', COALESCE(contacts.title, ''), ' ', COALESCE(contacts.first_name, ''), ' ', COALESCE(contacts.last_name, '')) as name, organisations.name as org_name")
                    ->leftJoin('organisations', 'organisations.id', '=', 'contacts.organisation_id')
                    ->orderByRaw('organisations.name ASC, contacts.last_name ASC');
            },
            'searchable_attributes' => ['first_name', 'last_name'],
            'distinct' => true,
        ]);
    }
}
```

It will not show the exact text you want, but it will help in debugging the issue. I will try to find my sample and check the exact cause.

**Original response**

Do you have the `FetchOperation` in your CRUD? It seems to me that the fetch operation is implemented manually and because of this it is missing the `/{id}` logic for when a record is already entered. Please check https://backpackforlaravel.com/docs/5.x/crud-fields#load-entries-from-ajax-calls-using-the-fetch-operation-1 once more.
